src/store/actions/authActions.js

```diff
@@ -1,28 +1,43 @@
 import * as ActionType from "./index";
 import axios from "axios";
-import UrlJoin from "url-join"
+import UrlJoin from "url-join";
+import { clearHackDetails } from './hackathonActions';
+const url = UrlJoin(process.env.REACT_APP_API_URL, "auth");
 
-const url = UrlJoin(process.env.REACT_APP_API_URL,"auth");
-
-
-const loginOkay = userId => ({ type: ActionType.LOGIN, payload: { userId: userId, status: "SUCCESS"} });
-const loginingIn = () => ({type: ActionType.LOGGING_IN, payload: "PENDING" })
-const loginFailed = () => ({type: ActionType.LOGGING_IN, payload: "FAILED" })
+const loginOkay = (userId, jwt) => ({
+  type: ActionType.LOGIN,
+  payload: { userId: userId, status: "SUCCESS", jwt: jwt },
+});
+const loggingIn = () => ({ type: ActionType.LOGGING_IN, payload: "PENDING" });
+const loginFailed = () => ({ type: ActionType.LOGGING_IN, payload: "FAILED" });
+const SessionExpired = () => ({ type: ActionType.LOGGING_IN, payload: "EXPIRED" });
+const logginOut = () => ({ type: ActionType.LOGGING_OUT });
 
 export const login = (email, password, history) => {
-
   return async (dispatch, getState) => {
-    dispatch(loginingIn())
+    dispatch(loggingIn());
     try {
       let result = await axios.post(url, {
         email,
         password
       });
       const { userId } = result.data;
-      dispatch(loginOkay(userId));
-      history.push('/hacks')
+      const jwt = result.data.AuthenticationResult.AccessToken;
+      dispatch(loginOkay(userId, jwt));
+      history.push("/");
     } catch (err) {
-      dispatch(loginFailed())
+      dispatch(loginFailed());
+    }
+  };
+};
+
+export const logout = (history,expired) => {
+  return async (dispatch, getState) => {
+    history.push("/login");
+    dispatch(clearHackDetails())
+    dispatch(logginOut());
```

Comment:
Is there a way to actually invalidate the token in cognito? if someone logs out, you should invalidate tokens so even if someone nefarious got access to the token they would no longer be able to use it.

Reply:
yeah, we will create another end point for that, great tip. Thank you

Reply:
Run into a problem doing this, can logout user with the Cognito **AdminUserGlobalSignOut**, however currently Cognito doesn't support invalidating the access token. That means current user is logged out, his session is wiped but his access is token is still valid and will pass through our checkAuth middleware. Right now research if there's a way to call Cognito method to check if user is still logged in after jwt.verify()

https://github.com/aws-amplify/amplify-js/issues/3435#issuecomment-597851340